src/Advection/vector_invariant_advection.jl

```diff
@@ -7,21 +7,24 @@ struct EnstrophyConservingScheme{FT} <: AbstractAdvectionScheme{1, FT} end
 EnergyConservingScheme(FT::DataType = Float64)    = EnergyConservingScheme{FT}()
 EnstrophyConservingScheme(FT::DataType = Float64) = EnstrophyConservingScheme{FT}()
 
-struct VectorInvariant{N, FT, Z, ZS, V, D, M} <: AbstractAdvectionScheme{N, FT}
-    vorticity_scheme  :: Z  # reconstruction scheme for vorticity flux
-    vorticity_stencil :: ZS # stencil used for assessing vorticity smoothness
-    vertical_scheme   :: V  # stencil used for assessing divergence smoothness
-    upwinding         :: D  # treatment of upwinding for divergence flux and kinetic energy gradient
+struct VectorInvariant{N, FT, Z, ZS, V, K, D, U, M} <: AbstractAdvectionScheme{N, FT}
+    vorticity_scheme   :: Z  # reconstruction scheme for vorticity flux
+    vorticity_stencil  :: ZS # stencil used for assessing vorticity smoothness
+    vertical_scheme    :: V  # recontruction scheme for vertical advection
+    ke_gradient_scheme :: K  # reconstruction scheme for kinetic energy gradient
+    divergence_scheme  :: D  # reconstruction scheme for divergence flux
+    upwinding          :: U  # treatment of upwinding for divergence flux and kinetic energy gradient
```

Comment:
There are changes to WENO vector invariant in this PR?

Reply:
It does not really change the default behavior, I introduced some flexibility to allow different schemes for the divergence flux and the kinetic energy gradient